### PR TITLE
Bug: Student received submission confirmation but the responses are not recorded #6643

### LINKED ISSUE #6643: Bug: Student received submission confirmation but the responses are not recorded
A user reported that one student submitted responses (and received a confirmation email) but the responses are missing from the results. 

v5.94

### PR DESCRIPTION
**Pull request by xpdavid**
Fixes #6643 

Details discussion of the bug is in the issue #6643 .

**Outline of Solution**

> 1. Disable the submission button in feedback submission page when user clicks it (to prevent from clicking it again).
> 2. It is not testable to test the behaviour above. I write JS test for the function (`addLoadingIndicator()`) used to disable the button to somehow test the behaviour indirectly.

**Demo**
demo

### CODE REVIEW

File: src/main/webapp/dev/allJsUnitTests.jsp [hunk 1]
```diff
@@ -298,6 +298,7 @@
             </tbody>
         </table>
     </div>
+    <button id="loading-indicator"></button>
```
Comment (li-kai) [inline src/main/webapp/dev/allJsUnitTests.jsp]: It's not advisable to keep adding new html unless it is really needed.

Take a look at using [fixtures](http://qunitjs.com/cookbook/#keeping-tests-atomic) to inject the button instead.

File: src/main/webapp/dev/CommonJsTest.js [hunk 2]
```diff
@@ -232,6 +232,24 @@ QUnit.test('checkEvaluationForm()', function(assert) {
     assert.expect(0);
 });
 
+QUnit.test('addLoadingIndicator()', function(assert) {
+    var $button = $('#loading-indicator');
+    addLoadingIndicator($button, 'Loading');
```
Comment (li-kai) [inline src/main/webapp/dev/CommonJsTest.js]: You might want to move the magic string `'Loading'` to a variable.

File: src/main/webapp/dev/CommonJsTest.js [hunk 3]
```diff
@@ -232,6 +232,24 @@ QUnit.test('checkEvaluationForm()', function(assert) {
...
+
+    assert.equal($button.text(), 'Loading', 'Button text changes to loading');
+    assert.equal($button.find('img').attr('src'), '/images/ajax-loader.gif', 'Loading gif appended');
+    assert.ok($button.is(':disabled'), 'Button disabled');
+});
+
+QUnit.test('removeLoadingIndicator()', function(assert) {
+    var $button = $('#loading-indicator');
+    removeLoadingIndicator($button, 'Complete');
```
Comment (li-kai) [inline src/main/webapp/dev/CommonJsTest.js]: Same case here for `'Complete'`

File: src/main/webapp/js/feedbackSubmissionsEdit.js [hunk 4]
```diff
@@ -767,6 +768,11 @@ function reenableFieldsForSubmission() {
     $(':disabled').prop('disabled', false);
 }
 
+function disableSubmissionButton() {
```
Comment (li-kai) [inline src/main/webapp/js/feedbackSubmissionsEdit.js]: I personally don't think 'SLAP' is very applicable to this function.

Two different suggestions
1. `addLoadingIndicator($('#response_submit_button'), '')` could be a single line with a comment.
2. move this function closer to its counterpart. 700 lines separation is a little excessive.

File: src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/feedbackSubmissionForm.tag [hunk 5]
```diff
@@ -41,13 +41,12 @@
...
                 Send me a confirmation email
-                <input type="submit" class="btn btn-primary center-block margin-top-7px"
+                <button type="submit" class="btn btn-primary center-block margin-top-7px"
                        id="response_submit_button" data-toggle="tooltip"
                        data-placement="top" title="<%= Const.Tooltips.FEEDBACK_SESSION_EDIT_SAVE %>"
-                       value="Submit Feedback"
                        <c:if test="${data.preview or (not data.submittable)}">
                            disabled style="background: #66727A;"
-                       </c:if>>
+                       </c:if>>Submit Feedback</button>
```
Comment (li-kai) [inline src/main/webapp/WEB-INF/tags/shared/feedbackSubmissionEdit/feedbackSubmissionForm.tag]: Would be nice to move the addition to a new line instead so it's easier to follow the hierarchy.

```
</c:if>>Submit Feedback</button>
```
to
```
<button type="submit"
        ....
        <c:if test="${data.preview or (not data.submittable)}">
            disabled style="background: #66727A;"
        </c:if>>
    Submit Feedback
</button>
```

**Comment by xpdavid** [comment]
@li-kai Ready for review.

**Comment by junhaoyap** [comment]
cc @damithc

**Comment by damithc** [comment]
Thanks for fixing this @xpdavid Just now another user complained about a similar issue :-p

**Pull request merged**